### ISSUE TITLE
Flag to override DB schema per query

### ISSUE THREAD

**Issue by neilotoole** [neilotoole/sq#270]
**Is your feature request related to a problem? Please describe.**

See this [comment](https://github.com/neilotoole/sq/issues/3#issuecomment-1598622940).

Given an existing source, it would be handy to be able to switch the schema on a per-query basis.

For example, given a source `@db1`:

```@shell
$ sq add 'postgres://sakila:p_ssW0rd@localhost/sakila' --handle @db1
```

It should be possible to switch the schema for a single query, e.g.

```shell
$ sq --schema customer1 '.customers'
```

This would probably work by mutating the source location before execution, such that the effective location would be:

```
postgres://sakila:p_ssW0rd@localhost/sakila?search_path=customer1
```

**Comment by neilotoole** [comment]
This has been addressed in [v0.43.0](https://github.com/neilotoole/sq/releases/tag/v0.43.0).

@c-nv-s: I think these changes should help with the friction described in this [comment](https://github.com/neilotoole/sq/issues/3#issuecomment-1598622940)? I'd love some feedback.

**Comment by c-nv-s** [comment]
thanks neil! so far this is just wonderful. can't see any bugs with the feature.
has made sq soo much more flexible

**Comment by neilotoole** [comment]
Thanks @c-nv-s, I much appreciate the feedback! Note as well that the `--src.schema` flag has got shell/tab completion, let me know if that works for you as expected.

You other feature request, #143 , is slowly bubbling toward the top of the list 😉

**Comment by c-nv-s** [comment]
the tab completion for ``--src.schema`` doesn't work for me likely because I have another script that takes over my shell autocomplete and git folder status reporting etc. I'll get around to checking it in a vanilla bash terminal another time to confirm.
for the meantime I just get a list of the available schemas with ``sq sql --exec  'select *from pg_namespace where nspowner = current_user::regrole;'`` then select from there as needed.

I am eternally grateful for this tool, and once ticket feature 143 is in it will literally be like a christmas gift for my workflow, I will likely abandon the ``psql`` cli completely.

**Comment by neilotoole** [comment]
@c-nv-s Well... if you haven't the experienced the shell completion yet, it truly will be like christmas. The completion is... extensive. I have a secret goal of making `sq`'s shell completion the best of any CLI anywhere. I like to think it's already pretty close. Almost every command/flag has some form of completion. I implore you, give it a try! 🙏

**Comment by neilotoole** [comment]
@c-nv-s: one more thing... The just-released [`v0.44.0`](https://github.com/neilotoole/sq/releases/tag/v0.44.0) has got a new mode for [`sq inspect`](https://sq.io/docs/inspect) that lists [schemas](https://sq.io/docs/inspect#schemata) (and [catalogs](https://sq.io/docs/inspect#catalogs)). It might be of use for your workflow. See the [docs](https://sq.io/docs/inspect#schemata).
